### PR TITLE
Command package refactor

### PR DESCRIPTION
**Pull request by adamgillessen**
So the main bones of the change:
- Changed the command functions so that they don't have to worry about sending stuff through discord
- Added unit tests to all commands
- Cleaned up the alias code
- Other code spring cleaning

It's best read as a new piece of code as opposed to reading the changes I think. The changes are pretty drastic and a lot of file moves were done.

### CODE REVIEW

File: commands/aliases.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+				}
+			}
+			p.Add(embed)
+		}
+
+		p.SetPageFooters()
+		p.Loop = true
+		p.ColourWhenDone = 0xFF0000
+		p.DeleteReactionsWhenDone = true
+		p.Spawn()
```
Comment (Strum355) [inline commands/aliases.go]: `p.Spawn()` returns an error, so we could do `return "", p.Spawn()`. I also feel like `p.Widget.Timeout = time.Minute * 5`. If i remember correctly, they never actually timed out and this may be the reason why

File: .drone.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ pipeline:
   unit_tests:
     image: golang:latest
     commands:
```
Comment (Strum355) [inline .drone.yml]: Add in the linting stage :D

File: commands/aliases.go [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+			command: func(_ context.Context, _ []string) (string, error) {
+				return aliasValue, nil
+			},
+		}
+	}
+	return commMap, nil
+}
+
+// aliasCommand sets string => string shortcut that can be called later to print a value
+func aliasCommand(ctx context.Context, args []string) (string, error) {
```
Comment (Strum355) [inline commands/aliases.go]: This function could use comments at each `if len(args)` to highlight what the course of action will be

File: commands/aliases.go [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+			},
+		})
+
+		for aliasName, aliasValue := range aliasMap {
+			embed := &discordgo.MessageEmbed{
+				Title:       aliasName,
+				Description: aliasValue,
+			}
+
+			resp, err := http.Head(aliasValue)
```
Comment (Strum355) [inline commands/aliases.go]: Evan made a point that we could pre-compute this, as doing it like this makes the listing slow as we're checking everything. Maybe you wanna do this? Else we can do it at a later stage

File: commands/aliases.go (hunk 3 above)
Comment (Strum355) [reply commands/aliases.go]: Found it a lil hard to follow, not being able to concentrate fully, just a minor thing :P

File: commands/aliases.go (hunk 1 above)
Comment (adamgillessen) [reply commands/aliases.go]: Changed to 
```golang
p.SetPageFooters()
p.Loop = true
p.ColourWhenDone = 0xFF0000
p.DeleteReactionsWhenDone = true
p.Widget.Timeout = 2 * time.Minute
if err := p.Spawn(); err != nil {
	return "", fmt.Errorf("Failed to spawn paginator: %s", err)
}
return "", nil
```
The timeout will now be 2 mins. I wrapped the error so that it has context. Returning an error like that is rarely a good idea.

File: commands/aliases.go (hunk 3 above)
Comment (adamgillessen) [reply commands/aliases.go]: Done :smile:

File: commands/aliases.go (hunk 4 above)
Comment (adamgillessen) [reply commands/aliases.go]: That's a good point. It's a change that probably doesn't fit in this PR. It will require changing the way aliases are stored. Not going to do it now but will do it in a different PR.

File: .drone.yml (hunk 2 above)
Comment (adamgillessen) [reply .drone.yml]: Done :smile:

File: commands/aliases.go (hunk 4 above)
Comment (Strum355) [reply commands/aliases.go]: Agreed :ok_hand:
